modules/miner/blockmanager.go

```diff
@@ -117,13 +149,28 @@ func (m *Miner) SubmitHeader(bh types.BlockHeader) error {
 	lookupBH := bh
 	lookupBH.Nonce = zeroNonce
 	lockID := m.mu.Lock()
-	b, exists := m.blockMem[lookupBH]
-	m.mu.Unlock(lockID)
-	if !exists {
+	b, bExists := m.blockMem[lookupBH]
+	randTxn, txnExists := m.randTxnMem[lookupBH]
+
+	if !bExists || !txnExists {
+		m.mu.Unlock(lockID)
 		err := errors.New("block header returned late - block was cleared from memory")
 		m.log.Println("ERROR:", err)
 		return err
 	}
-	b.Nonce = bh.Nonce
-	return m.SubmitBlock(b)
+	// Reset block memory
+	m.memProgress = 0
```

Comment:
I guess i was thinking that since a block was submitted, the miner should start over since no old blocks will be valid any more. But along that line of thinking, I should have cleared `blockMem`, `randTxnMem` and `headers`

Should I worry about deleting old block memory?